File: 백준/Bronze/2231. 분해합/분해합.py
import math
import heapq
from collections import deque
from itertools import permutations, combinations
import sys
input = sys.stdin.readline


# 분해합이 들어오면 
# 그 분해합을 가지는 가장 작은 자연수를 출력하기 

if __name__ == "__main__":
    N = int(input())
    result = 0

    for i in range(N+1):
        decomposition_sum = i + sum(int(num) for num in str(i))

        if decomposition_sum == N:
            result = i
            break
    
    print(result)

File: 백준/Bronze/2231. 분해합/분해합.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <math.h>
#include <ctype.h>
#include <limits.h>

#define TRUE 1 
#define FALSE 0


int main() {
    int N;
    scanf("%d", &N);

    for(int i = 0; i < N; i++){
        int sum = 0;
        int num = i;

        sum += i;
        while(num > 0){
            sum += num % 10;
            num /= 10;
        }

        if(sum == N){
            printf("%d\n", i);
            return 0;
        }
    }

    printf("0\n");
    return 0;
}